### PR TITLE
Don't require a returns section for `NoReturn`

### LINKED ISSUE #55: Support return type `NoReturn`
Currently if a function uses the return type `typing.NoReturn` a `Returns:` section is still required in the docstring, which doesn't make a lot of sense.

```py
from typing import NoReturn


def f() -> NoReturn:
    """This function never returns.

    Raises:
        Exception: Always raises an exception.
    """
    raise Exception("Error")
```

```console
$ pydoclint --style=google --check-return-types=False --arg-type-hints-in-docstring=False --arg-type-hints-in-signature=True noreturn.py 
Skipping files that match this pattern: \.git|\.tox
noreturn.py
noreturn.py
    4: DOC201: Function `f` does not have a return section in docstring 
```

### PR DESCRIPTION
**Pull request by llucax**
When a function has a return type of `NoReturn` it means it will never return, so the documentation doesn't really need to specify a return.

The option `--require-return-section-when-returning-none` was renamed to `--require-return-section-when-returning-nothing` to highlight it is more generic now, as it will also require a returns section for `NoReturn`. The old option is kept as a deprecation error.

Fixes #55.

### CODE REVIEW

File: pydoclint/utils/return_yield_raise.py [hunk 1]
```diff
@@ -27,6 +27,19 @@ def _isNone(node: ast.AST) -> bool:
...
+def isReturnAnnotationNoReturn(node: FuncOrAsyncFuncDef) -> bool:
+    """Check whether the return type annotation if `NoReturn`"""
+    if node.returns is None:
+        return False
+
+    returnAnnotation: str = unparseAnnotation(node.returns)
+    return returnAnnotation.startswith('NoReturn')
+
+
+def _isNoReturn(node: ast.AST) -> bool:
```
Comment (jsh9) [inline pydoclint/utils/return_yield_raise.py:39]: This function doesn't seem to be used anywhere?

File: pydoclint/visitor.py [hunk 2]
```diff
@@ -478,8 +485,11 @@ def checkReturns(  # noqa: C901
 
             if (
                 returnSec == []  # no return section in docstring
-                and returnAnno.annotation == 'None'  # `-> None` in signature
-                and not self.requireReturnSectionWhenReturningNone
+                and (
+                    returnAnno.annotation == 'None'  # `-> None` in signature
+                    or returnAnno.annotation == 'NoReturn'  # `-> NoReturn` in signature
+                )
```
Comment (jsh9) [inline pydoclint/visitor.py:491]:
```suggestion
                # `-> None` or `-> NoReturn` in signature
                and returnAnno.annotation in {'None', 'NoReturn'}
```

File: pydoclint/utils/return_yield_raise.py (hunk 1 above)
Comment (jsh9) [reply pydoclint/utils/return_yield_raise.py:39]: And comparing the implementation of this function and that of `isReturnAnnotationNoReturn()`, it seems that the implementation of this function is better?

**Comment by jsh9** [comment]
Hi @llucax , thank you so much for finding this issue and submitting this PR!

The code changes generally look good, and I think the renaming from "none" and "nothing" is very appropriate!

I left a few comments above, basically some curious questions.  Would you mind taking a look at them?

File: pydoclint/utils/return_yield_raise.py (hunk 1 above)
Comment (llucax) [reply pydoclint/utils/return_yield_raise.py:39]: Hi @jsh9!

> This function doesn't seem to be used anywhere?

Yeah, good catch. This is a leftover from a version that didn't work. I should (maybe?) remove it.

> And comparing the implementation of this function and that of `isReturnAnnotationNoReturn()`, it seems that the implementation of this function is better?

It looks better but it didn't work for me. I'm not familiar with the `ast` stuff here, so not sure if there is a way to make it work, but when I tried it it seems like `NoReturn` is not a `ast.Constant` but a `ast.Name` IIRC., this is why I had to go with the string comparison. I think I could also just do `returnAnnotation == 'NoReturn'`, the `startwidth()` was based on the generator stuff but in this case I think it should only match to `NoReturn`.

**Comment by llucax** [comment]
Hi @jsh9, I updated the PR with your suggestions and comments. Let me know if you think `isReturnAnnotationNoReturn()` can be improved.

**Comment by jsh9** [comment]
Hi @llucax , I think `isReturnAnnotationNoReturn()` is good.  Let me merge it now.

Thanks again for the contribution!

**Pull request merged**